src/lib/transit/station/spatial-index.ts

```
@@ -0,0 +1,146 @@
+// Import GeoJSON namespace
+import type { GeoJSON } from 'geojson';
+
+export class SpatialIndex<T extends { coordinates: Coordinates }> {
+  private items: T[] = [];
+  // Fix the type to accept both Point and MultiPoint
+  private pointFeatures: GeoJSON.FeatureCollection<
+    GeoJSON.Point | GeoJSON.MultiPoint,
+    { index: number }
+  > | null = null;
+
+  constructor() {}
```

Biome (1.9.4)

[error] This constructor is unnecessary.

Unsafe fix: Remove the unnecessary constructor.


(lint/complexity/noUselessConstructor)